File: deploy/openvino/lenet.py
import cv2
import numpy as np
from openvino.inference_engine import IECore


img = cv2.imread("10.png", 0)
blob = cv2.dnn.blobFromImage(img, 1/255., size=(28,28), swapRB=True, crop=False)
ie = IECore()
#net = ie.read_network(model="lenet.onnx")
net = ie.read_network(model="lenet/lenet_fp16.xml", weights="lenet/lenet_fp16.bin")
exec_net = ie.load_network(network=net, device_name="CPU")
input_layer = next(iter(net.input_info))
infer_request_handle = exec_net.start_async(request_id=0, inputs={input_layer: blob})
if infer_request_handle.wait(-1) == 0:
    output_layer = infer_request_handle._outputs_list[0]
    outputs = infer_request_handle.output_blobs[output_layer] 
    #outputs = infer_request_handle.output_blobs["23"]
    print(np.argmax(outputs.buffer))

File: deploy/openvino/lenet.cpp
#include <iostream>
#include <opencv2/opencv.hpp>
#include <openvino/openvino.hpp>


int main(int argc, char* argv[])
{
	ov::Core core;
	//auto model = core.compile_model("lenet.onnx", "CPU");             
	auto model = core.compile_model("lenet/lenet_fp16.xml", "CPU");
	auto iq = model.create_infer_request();
	auto input = iq.get_input_tensor(0);
	auto outputs = iq.get_output_tensor(0);
	input.set_shape({ 1, 1, 28, 28 });
	float* inputs = input.data<float>();

	cv::Mat image = cv::imread("10.png", 0);
	image.convertTo(image, CV_32F, 1.0 / 255);
	for (int i = 0; i < 28; i++)
	{
		for (int j = 0; j < 28; j++)
		{
			inputs[i * 28 + j] = image.at<float>(i, j);
		}
	}

	iq.infer();

	float* pred = outputs.data<float>();
	int predict_label = std::max_element(pred, pred + 10) - pred;
	std::cout << predict_label << std::endl;

	return 0;
}